### PR TITLE
Reduce sleep to make tests more fast

### PR DESCRIPTION
**Pull request by vitormattos**
I used microseconds to enable testing at shorter intervals.
On the first run there is no sleep.
In the second run, sleep is half a second.
Repetitions are made for 1 and a half seconds.

### CODE REVIEW

**Comment by donatj** [comment]
You can see from the failed integration tests that `isRunning` doesn't mean it's ready to accept incoming connections, just that the process id exists. 

`isRunning` will return _true_ at _any point_ after the call to `exec` on line 79 unless the server otherwise crashes.

Being able to tell when the server is actually able to accept connections is a bigger ask.

I'd love to see a reduction in the time, but it would require a larger more systemic change.

**Comment by vitormattos** [comment]
Sorry for not checking the continuous integration output, I only tested it on the application I'm using your package.

I added a health check.

With this new approach sleep can even be removed or further reduced and keep the health check in a loop for 1 second. What do you think?

**Comment by donatj** [comment]
I pushed to your branch what I think is probably a simpler solution. Thoughts?

File: src/MockWebServer.php [hunk 1]
```diff
@@ -86,7 +86,15 @@ public function start() {
 			throw new Exceptions\ServerException("Error starting server, received '{$this->pid}', expected int PID");
 		}
 
-		sleep(1); // just to make sure it's fully started up, maybe not necessary
+		for( $i = 0; $i <= 20; $i++ ) {
+			usleep(100000);
```
Comment (donatj) [inline src/MockWebServer.php:90]: I sleep 100ms from the start because on my decently fast machine _every single execution_ for otherwise was a retry.

Comment (vitormattos) [reply src/MockWebServer.php:90]: Excellent approach! It's a good Kabbalistic number. lol

**Comment by donatj** [comment]
@vitormattos I just want to say thank you for this!

I just tried it on a big project I personally use mock-webserver on and it knocked phpunit from ~15 seconds down to 1.75 total, so that's a pretty decent win.

**Comment by donatj** [comment]
This has been tagged in release [v2.3.0](https://github.com/donatj/mock-webserver/releases/tag/v2.3.0)

**Comment by vitormattos** [comment]
This is the power of FLOSS projects!

**Pull request merged**